src/d3a/d3a_core/sim_results/plotly_graph.py

```diff
@@ -0,0 +1,584 @@
+
+    @staticmethod
+    def modify_time_axis(data: dict, title: str):
+        """
+        Changes timezone of pendulum x-values to 'UTC' and determines the list of days
+        in order to return the time_range for the plot
+        """
+        day_set = set()
+        for di in range(len(data)):
+            time_list = data[di]["x"]
+            for ti in time_list:
+                day_set.add(pendulum.datetime(ti.year, ti.month, ti.day, tz=TIME_ZONE))
+
+        day_list = sorted(list(day_set))
+        if len(day_list) == 0:
+            raise ValueError("There is no time information in plot {}".format(title))
+
+        start_time = pendulum.datetime(day_list[0].year, day_list[0].month, day_list[0].day,
+                                       0, 0, 0, tz=TIME_ZONE)
+        end_time = pendulum.datetime(day_list[-1].year, day_list[-1].month, day_list[-1].day,
+                                     23, 59, 59, tz=TIME_ZONE)
+
+        return [start_time, end_time], data
+
+    @classmethod
+    def plot_bar_graph(cls, barmode: str, title: str, xtitle: str, ytitle: str, data, iname: str):
+        try:
+            time_range, data = cls.modify_time_axis(data, title)
+        except ValueError:
+            return
+
+        layout = cls.common_layout(barmode, title, ytitle, xtitle, time_range)
+
+        fig = go.Figure(data=data, layout=layout)
+        py.offline.plot(fig, filename=iname, auto_open=False)
+
+    @classmethod
+    def plot_line_graph(cls, title: str, xtitle: str, ytitle: str, data, iname: str, xmax: int):
+        layout = cls.common_layout("group", title, ytitle, xtitle, [0, xmax])
+
+        fig = go.Figure(data=data, layout=layout)
+        py.offline.plot(fig, filename=iname, auto_open=False)
+
+    def arrange_data(self):
+        try:
+            self.dataset[self.key]
+        except KeyError:
+            pass
+        else:
+            for ii, ki in enumerate(self.dataset[self.key]["seller"]):
+                if ki in self.trade_history.keys():
+                    self.trade_history[ki] += abs(self.dataset[self.key]["energy [kWh]"][ii])
+                else:
+                    self.trade_history[ki] = abs(self.dataset[self.key]["energy [kWh]"][ii])
+
+    def plot_pie_chart(self, title, filename):
+        fig = {
+            "data": [
+                {
+                    "values": list(),
+                    "labels": list(),
+                    "type": "pie"
+                }],
+            "layout": {
+                "title": title,
+                "font": {"size": 16
+                         }
+            }
+        }
+        for key, value in self.trade_history.items():
+            fig["data"][0]["values"].append(value)
+            fig["data"][0]["labels"].append(key)
+
+        py.offline.plot(fig, filename=filename, auto_open=False)
+
+    @classmethod
+    def _plot_line_time_series(cls, device_dict, var_name):
```

Comment:
Same for this, I think it is better this method to be public since it is called from outside of this class